library/src/test/java/kwasm/format/text/FunctionTypeTest.kt

```diff
@@ -29,48 +33,81 @@ import org.junit.runners.JUnit4
 class FunctionTypeTest {
     @Test
     fun parseValidFunctionType_OneParamOneReturn() {
-        val onlyParam = listOf(Param(Identifier.Local("\$val1"), ValueTypeEnum.I32))
-        val onlyReturnType = listOf(ValueTypeEnum.I32)
-        val expected = FunctionType(onlyParam, onlyReturnType)
-        val actual = Type.FunctionType("(func (param \$val1 i32) (result i32))")
-        assertThat(actual.value).isEqualTo(expected)
+        val onlyParam = listOf(Param(kwasm.ast.Identifier.Local("\$val1"), ValueType(ValueTypeEnum.I32)))
+        val onlyReturnType = listOf(Result(ValueType(ValueTypeEnum.I32)))
+        val expected = ParseResult(FunctionType(onlyParam, onlyReturnType), 12)
+        val actual = listOf(
+            Paren.Open(), Keyword("func"), Paren.Open(),
+            Keyword("param"), Identifier("\$val1"), Keyword("i32"),
+            Paren.Closed(), Paren.Open(), Keyword("result"), Keyword("i32"),
+            Paren.Closed(), Paren.Closed()
+        ).parseFunctionType(0)
```

Comment:
It'd probably be cleaner/easier to read these tests (for all in the PR) if you use the tokenizer, and just feed in Wasm text format stuff.

Reply:
done